.github/actions/changes/get_pr_info.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+# Copyright (c) The Diem Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+function echoerr() {
+  cat <<< "$@" 1>&2;
+}
+
+#Check prerequists.
+function check_command() {
+    for var in "$@"; do
+      if ! (command -v "$var" >/dev/null 2>&1); then
+          echoerr "This command requires $var to be installed"
+          exit 1
+      fi
+    done
+}
+check_command jq curl tr echo getopts git
+
+function usage() {
+  echoerr -u github username \(only needed for private repos\).
+  echoerr -p github password \(only needed for private repos\).
+  echoerr -w workflow file name, used to find old runs in pushed updates for git base revision comparison.
+  echoerr -b if we are using bors, bors-ng, etc, use pr commit messages to determine target branch for merge if in \'auto\' branch.
+  echoerr -d print debug messages.
+  echoerr -h this message.
+  echoerr output will be echoed environment variables
+  echoerr GITHUB_SLUG, the repo that has been written to, or a pr is attempting to write to.
+  echoerr TARGET_BRANCH, the git branch that a pr is attempting to write to, if this build is for a pr.
+  echoerr BASE_GITHASH, if the build is pr, the merge-base, if a push, the last build of this branch using the -g parameter.
+  echoerr CHANGED_FILE_OUTPUTFILE the path to a temp file which is the delta between the BASE_GITHASH and the current build\'s HEAD
+  echoerr if the changed file set cannot be calculated for any reason the script will return -1, after echoing out all the
+  echoerr variables it could calculate.
+}
+
+#Optional: if a private repo
+GITHUB_USERNAME=;
+GITHUB_PASSWORD=;
+BORS=false
+DEBUG=false
+WORKFLOW_FILE=;
+
+while getopts 'u:p:g:bdw:h' OPTION; do
+  case "$OPTION" in
+    u)
+      GITHUB_USERNAME="$OPTARG"
+      ;;
+    p)
+      GITHUB_PASSWORD="$OPTARG"
+      ;;
+    b)
+      BORS=true
+      ;;
+    d)
+      DEBUG=true
+      ;;
+    w)
+      WORKFLOW_FILE="$OPTARG"
+      ;;
+    ?)
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+if [[ -z "${WORKFLOW_FILE}" ]]; then
+  echoerr Workflow file must be sepecified.
+  usage
+  exit 1
+fi
+
+#calculate the login for curl
```

Comment:
This logic all needs refactoring.   Not happy.  But it works in prs and pushs, need to test in bors, but that section is from the equivalent file in circleci.    Called only by the changes action.